Automattic/pocket-casts-ios
# Patron: Add Initial Server Support

This adds the initial support for the server changes which adds a new "Tier" field to the SubscriptionStatus request. 

This also replaces the usage of determining which level of subscription the user has from the SubscriptionType to the new SubscriptionTier. 

I also added a fallback to fill in the SubscriptionTier if the server isn't returning it yet so I can unify the usage of Tier while it's not readily available. 

## To test

**Note:** While the endpoint changes are available in staging they are not currently working yet. If you want to test the flow fully message me and I can setup an env for you to test.

1. Launch the app
2. Sign into an account with Plus
3. Go to the Profile tab
4. ✅ Verify your plus is recognized and displays the correct information
5. Tap Account
6. ✅ Verify plus is recognized and displays the correct information
7. Go to the Podcasts tab
8. ✅ Verify you can make a new folder
9. Use the Profile > Cog > Developer menu to switch to Paid Patron
10. Go back to the Profile tab
11. ✅ Verify the Patron status is reflected


## Checklist
